https://issues.apache.org/jira/browse/EAGLE-104

pmujumdar opened the pull request:
EAGLE-104: Fix the unit test TestHDFSSecuritylogParser
The problem is that the UGI parsing method is not handling the simple authentication scenario where user name is not a kerberos principal. The patch is a minor refactoring of the method that parses the ugi.
Added new testcases to validate the changes.  Ran unit tests, no failures in eagle-security.

haoch commented:
@sunlibin please help review this pull request.

yonzhang commented:
I run this unit test in my local and observed the following errors
-------------------------------------------------------------------------------
Test set: org.apache.eagle.security.TestHDFSSecuritylogParser
-------------------------------------------------------------------------------
Tests run: 3, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 0.404 sec <<< FAILURE!
testSimpleAuth(org.apache.eagle.security.TestHDFSSecuritylogParser)  Time elapsed: 0.002 sec  <<< ERROR!
java.lang.StringIndexOutOfBoundsException: String index out of range: -1
        at java.lang.String.substring(String.java:1911)
        at org.apache.eagle.security.util.LogParseUtil.parseUserFromUGI(LogParseUtil.java:39)
        at org.apache.eagle.security.securitylog.parse.HDFSSecurityLogParser.parseAttr(HDFSSecurityLogParser.java:78)
        at org.apache.eagle.security.securitylog.parse.HDFSSecurityLogParser.parse(HDFSSecurityLogParser.java:52)
        at org.apache.eagle.security.TestHDFSSecuritylogParser.verifyParserAttributes(TestHDFSSecuritylogParser.java:63)
        at org.apache.eagle.security.TestHDFSSecuritylogParser.testSimpleAuth(TestHDFSSecuritylogParser.java:38)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:606)
        at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:45)
        at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:15)
        at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:42)
        at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:20)
        at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:263)
        at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:68)
        at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:47)
        at org.junit.runners.ParentRunner$3.run(ParentRunner.java:231)
        at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:60)
        at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:229)
        at org.junit.runners.ParentRunner.access$000(ParentRunner.java:50)
        at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:222)
        at org.junit.runners.ParentRunner.run(ParentRunner.java:300)
        at org.apache.maven.surefire.junit4.JUnit4TestSet.execute(JUnit4TestSet.java:59)
        at org.apache.maven.surefire.suite.AbstractDirectoryTestSuite.executeTestSet(AbstractDirectoryTestSuite.java:120)
        at org.apache.maven.surefire.suite.AbstractDirectoryTestSuite.execute(AbstractDirectoryTestSuite.java:103)

sunlibin commented:
@pmujumdar, thanks for the fix, but please make parseUserFromUGI() without regular-expression, because it will affect our performance

pmujumdar commented:
@sunlibin Thanks for the feedback.
I see there are a bunch of regex checks already in that code path. But this is called on every log entry, fair enough. I updated the patch to restore the old code and added the missing case for non-kerberos principal.
Also since removed some of the trim() methods which are not really needed which should help the performance a bit more.
Rebased the branch and ran tests successfully in parent directory eagle-security. Fixed a couple of license header issues that hit on latest master.

yonzhang commented:
I have run that in my local, it is working fine. But Prasad can you please re-open this pull request with the title prefixed by EAGLE-104 as at backend some script runs based on this naming convention.
For example the pull request title can be "EAGLE-104 fix unit test bug in hdfs security log parser"
After that, I will merge this on-behalf-of you

pmujumdar commented:
ah ok. I didn't realize that. sorry about the trouble. 
will create a new PR.

pmujumdar commented:
Create new PR #61 per review feedback. Closing this request #43.

yonzhang commented:
Somehow I have merged this pull request, but this PR is not automatically closed.
